src/components/HOC/AuthRedirect.tsx

```diff
@@ -0,0 +1,23 @@
+import useUserInfo from "../../hooks/useUserInfo";
```

Comment:
This should be a generic auth HOC. It should take in component that is displayed when user is authed and redirect to log in page if not. https://nextjs.org/docs/api-reference/next/router